clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -762,15 +765,24 @@ public boolean isDone() {
     private static class InterceptorCallback<K, V> implements Callback {
```

Comment:
Not related to this patch, but would it make sense to rename this to something like CallbacksAndInterceptors, or something of that sort, as this is made up of user and interceptor callbacks,.